pkg/activator/handler/probe_handler_test.go

```diff
@@ -110,3 +111,42 @@ func mapToHeader(m map[string]string) http.Header {
 	}
 	return h
 }
+
+func BenchmarkProbeHandler(b *testing.B) {
+	examples := []struct {
+		label   string
+		headers http.Header
+	}{{
+		label:   "valid header",
+		headers: mapToHeader(map[string]string{network.ProbeHeaderName: activator.Name}),
+	}, {
+		label:   "invalid header",
+		headers: mapToHeader(map[string]string{network.ProbeHeaderName: "not-empty"}),
+	}, {
+		label:   "empty header",
+		headers: http.Header{},
+	}}
+
+	baseHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {})
+	handler := ProbeHandler{NextHandler: baseHandler}
+	req := httptest.NewRequest(http.MethodPost, "http://example.com", nil)
+
+	for i, e := range examples {
+		req.Header = e.headers
+		b.Run(fmt.Sprintf("%d-sequential", i), func(b *testing.B) {
+			resp := httptest.NewRecorder()
+			for j := 0; j < b.N; j++ {
+				handler.ServeHTTP(resp, req)
+			}
+		})
+
+		b.Run(fmt.Sprintf("%d-parallel", i), func(b *testing.B) {
+			b.RunParallel(func(pb *testing.PB) {
+				for pb.Next() {
+					resp := httptest.NewRecorder()
```

Comment:
You can probably move this above the for loop.

Reply:
If i move above the for loop it will give `concurrent map writes` for parallel scenario